domain-model/src/main/java/md/maib/retail/model/campaign/Campaign.java

```diff
@@ -26,12 +28,13 @@ public final class Campaign {
 
             private Collection<Rule> rules;
 
-            public boolean activate() throws Exception {
+
+            public boolean activate() throws CampaignActivationException {
                 if (state == CampaignState.DRAFT) {
                     this.state = CampaignState.ACTIVE;
                     return true;
                 } else {
-                    return false;
+                    throw new CampaignActivationException("Campaign cannot be activated because it is not in DRAFT state.");
```

Comment:
we use exceptions only for exceptional cases. E.g. can not connected to database or other IO exception
for business cases use boolean or Optional or Either

Reply:
i have to delete exception and to make return as a false?